cordformation/src/noderunner/kotlin/net/corda/plugins/NodeRunner.kt

```diff
@@ -102,27 +105,60 @@ private class NodeRunner(baseDebugPort: Int, baseMonitoringPort: Int) {
     }
 
     private fun startWindowed(command: List<String>, workingDir: File, nodeName: String): Process {
+
+        val createTempScriptFile = {
+            val tmpFile: File = File.createTempFile("corda_node_script_launcher", "")
+            BufferedWriter(OutputStreamWriter(FileOutputStream(tmpFile))).use {
+                //Start shell to keep window open unless java terminated normally or due to SIGTERM:
+                it.write("""
+                        #!/bin/sh
+                        set -v
+                        unlink $tmpFile
+                        cd "$workingDir"
+                        command -v rlwrap
+                        if [ -${'$'}? -eq 0 ]
```

Comment:
the `-` before the dollar sign is actually a typo, it doesn't cause any harm because `0` is `-eq` to `-0`, which makes sense, try to run
```
if [ -0 -eq 0 ]; then echo success; else echo fail; fi
```
to validate my claim..
btw I'm removing the dash in a following revision

Reply:
it could actually cause trouble in case `command` exited with a negative return value but that seems to be impossible
https://unix.stackexchange.com/questions/418784/what-is-the-min-and-max-values-of-exit-codes-in-linux